sfge/src/test/java/com/salesforce/rules/unusedmethod/BaseUnusedMethodTest.java

```diff
@@ -255,25 +259,37 @@ protected void assertExpectations(
             String entryMethod,
             Collection<String> usedMethodKeys,
             Collection<String> unusedMethodKeys) {
-        TestUtil.buildGraph(g, sourceCodes);
 
+        //TODO: This method attempts to simulate what AbstractRuleRunner.runRules does minus all the thread specific code
+        //      We should refactor the AbstractRuleRunner so that the Thread Specific parts can be injected as a strategy
+        //      and then for testing we would inject a strategy that executes without threads so that we can simply
+        //      call through to the top level rule runner here in the tests. Doing this refactoring would help us
+        //      prevent accidentally forgetting to update this code when changes occur somewhere deep down
+        //      inside of the runner's logic (like calling postProcess for example).
+
+        TestUtil.buildGraph(g, sourceCodes);
         RemoveUnusedMethod rule = RemoveUnusedMethod.getInstance();
         MethodVertex entryMethodVertex =
                 TestUtil.getMethodVertex(g, entryDefiningType, entryMethod);
         PathBasedRuleRunner runner =
                 new PathBasedRuleRunner(g, Lists.newArrayList(rule), entryMethodVertex);
-        // Violations aren't actually generated during the `runRules()` call.
         List<Violation> violations = new ArrayList<>(runner.runRules());
+        violations.addAll(rule.postProcess(g)); // Important! This is where the violations are created for this rule
+
         UsageTracker usageTracker = new UsageTracker();
         for (String usedMethodKey : usedMethodKeys) {
             assertTrue(
                     usageTracker.isUsed(usedMethodKey),
                     "Expected usage of method " + usedMethodKey);
         }
+
+        Set<String> actualViolationKeys = violations.stream()
+            .map(v -> v.getSourceDefiningType() + "::" + v.getSourceVertexName() + "::" + v.getSourceLineNumber() )
```

Comment:
Question: Why generate the actual keys with a different style and convert the expected keys to that style, instead of just generating the actual keys in the same style?

Reply:
I started with using '#' but I see many tests use '@'. I couldn't see why... i'll look into it more today.

Reply:
Ah I figured it out... the only case where '#' was used instead of '@' was in ConstructorsTest.java
   assertNoUsage(new String[] {sourceCode}, "MyClass", "entrypoint", "MyClass#<init>#2");
(that is... the test that was falsely passing before because of the way the test helper was written).
A simple update to this test, fixes this issue so I don't need to normalize.